### DISCUSSION TITLE
function filter VI

### DISCUSSION THREAD

**Discussion by jjbagus** [atralice/Curso.Prep.Henry#53] [Q&A]
Alguno me puede dar una mano para entender como funciona bien esto del cb en forEach? En las soluciones está planteada una resolución con bucle for, pero me pareció podía ser resuelto con el bucle foreach, pero no le encuentro la vuelta. Gracias!

function filter(array) {
  //Filtrar todos los elementos del array que comiencen con la letra "a".
  //Devolver un nuevo array con los elementos que cumplen la condición
  //Tu código:
  var newArr = []
    function Assrober (){
      if (this[0] === 'a'){
      newArr.push(this);
    }
  }
  array.forEach(Assrober());
  return newArr;
}

**Comment by atralice** [comment]
Buenas, 
podrías usar `forEach`, ya que tenes que invocar la `cb` en cada elemento del arreglo. Y según el resultado del `cb` pasandole cada elemento, vas a ir guardandolo en un arreglo o no.
Y retornas ese arreglo.
